node/src/actors/epoch_manager/mod.rs

```diff
@@ -146,14 +146,21 @@ impl EpochManager {
     fn time_to_next_checkpoint(&self) -> EpochResult<Duration> {
         // Get current timestamp and epoch
         let (now_secs, now_nanos) = get_timestamp_nanos();
-        let current_epoch = self.epoch_at(now_secs)?;
-
-        // Get timestamp for the start of next checkpoint
-        let next_checkpoint = self.epoch_timestamp(
-            current_epoch
-                .checked_add(1)
-                .ok_or(EpochManagerError::Overflow)?,
-        )?;
+        let current_epoch_res = self.epoch_at(now_secs);
+
+        let next_checkpoint =
+            if let Err(EpochManagerError::CheckpointZeroInTheFuture(zero)) = current_epoch_res {
```

Comment:
I think this would be more clear as a `match`

Reply:
I agree, but I was unsure of whether `clippy` would complain about using match for a simple two-clause conditional. If `clippy` is OK with it, it's a :+1: from me.

Reply:
clippy approves